File: Minimum Size Subarray Sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        slow=0
        fast=0
        summ=0
        minimumsize=float("inf")
        result=None
        while fast<len(nums):
            summ+=nums[fast]
            fast+=1
            if summ>=target:
                while summ>=target:
                    if minimumsize>fast-slow:
                        result=fast-slow
                        minimumsize=result
                    summ-=nums[slow]
                    slow+=1
        return result or 0

File: Java/Minimum Size Subarray Sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int left=0;
        int sum =0;
        int minvalue=Integer.MAX_VALUE;
        for(int right =0 ; right < nums.length ; right++){
            sum+=nums[right];
            while(sum>=target){
                minvalue=Math.min(minvalue,right-left+1);
                sum-=nums[left];
                left++;

            }
        }
        if(minvalue==Integer.MAX_VALUE){
            return 0;
        }
         
        return minvalue;
    }
}
